kong/router/expressions.lua

```diff
@@ -30,7 +30,9 @@ local function get_exp_and_priority(route)
   local protocols = route.protocols
 
   -- give the chance for http redirection (301/302/307/308/426)
-  if protocols and #protocols == 1 and protocols[1] == "https" then
+  if protocols and #protocols == 1 and
+    (protocols[1] == "https" or protocols[1] == "tls")
```

Comment:
Does HTTP redirection also work for TLS stream?

Reply:
Sorry. that comment may mislead you. but in stream subsystem will also need to do the same thing.

Reply:
@chronolaw Basically this condition allows HTTP requests to get proxied to HTTPS service. I don't think we need the same logic for stream.

Reply:
add a comment to explain it.